channel/consensus_channel/consensus_channel.go

```diff
@@ -143,38 +143,26 @@ var ErrInvalidDeposit = fmt.Errorf("unable to divert to guarantee: invalid depos
 var ErrInsufficientFunds = fmt.Errorf("unable to divert to guarantee: insufficient funds")
 var ErrDuplicateGuarantee = fmt.Errorf("duplicate guarantee detected")
 
-// DivertToGuarantee deducts g.amount from o.left's balance, and
-// adds g to o.guarantees
-func (o LedgerOutcome) DivertToGuarantee(p Add) (LedgerOutcome, error) {
-	g := p.Guarantee
-
-	if types.Gt(&p.LeftDeposit, &g.amount) {
-		return LedgerOutcome{}, ErrInvalidDeposit
-	}
-
-	if types.Gt(&g.amount, &o.left.amount) {
-		return LedgerOutcome{}, ErrInsufficientFunds
-	}
-
-	o.left.amount.Sub(&o.left.amount, &p.LeftDeposit)
-	rightDeposit := p.RightDeposit()
-	o.right.amount.Sub(&o.right.amount, &rightDeposit)
-
-	_, found := o.guarantees[g.target]
-	if found {
-		return LedgerOutcome{}, ErrDuplicateGuarantee
-	}
-	o.guarantees[g.target] = g
-
-	return o, nil
-}
-
 // Vars stores the turn number and outcome for a state in a consensus channel
 type Vars struct {
 	TurnNum uint64
 	Outcome LedgerOutcome
 }
 
+// clone returns a deep clone of v
+func (v Vars) clone() Vars {
```

Comment:
From [Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments#receiver-type):
"Don't mix receiver types. Choose either pointers or struct types for all available methods."

Later in this file, we have https://github.com/statechannels/go-nitro/blob/c3e14a8775b04d5bd54cf02d52537e1707f9937b/channel/consensus_channel/consensus_channel.go#L199

Reply:
7f43928c9d44a0634248726f6857b4b3ff5a9867 naturally addresses this